app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java

```diff
@@ -225,6 +254,37 @@ private void onEnterEntryResult(int resultCode, Intent data) {
         }
     }
 
+    private void onScanImageResult(int resultCode, Intent intent) {
+        if (resultCode == RESULT_OK) {
+            Uri inputFile = (intent.getData());
+            InputStream inputStream;
+            Bitmap bitmap;
+
+            try {
+                inputStream = getContentResolver().openInputStream(inputFile);
+                BitmapFactory.Options bmOptions = new BitmapFactory.Options();
+                bitmap = BitmapFactory.decodeStream(inputStream, null, bmOptions);
+
+                int[] intArray = new int[bitmap.getWidth() * bitmap.getHeight()];
+                bitmap.getPixels(intArray, 0, bitmap.getWidth(), 0, 0, bitmap.getWidth(), bitmap.getHeight());
+
+                LuminanceSource source = new RGBLuminanceSource(bitmap.getWidth(), bitmap.getHeight(), intArray);
+                BinaryBitmap binaryBitmap = new BinaryBitmap(new HybridBinarizer(source));
+
+                Reader reader = new MultiFormatReader();
+                Result result = reader.decode(binaryBitmap);
+
+                GoogleAuthInfo info = GoogleAuthInfo.parseUri(result.getText());
+                DatabaseEntry entry = new DatabaseEntry(info);
+
+                startEditProfileActivity(CODE_ADD_ENTRY, entry, true);
+            } catch (Exception e) {
```

Comment:
This would also catch any unchecked exceptions. Please make this a multi-catch statement that only catches the checked exceptions that the above code block can throw.